querying/project.go

```diff
@@ -0,0 +1,89 @@
+package querying
+
+import (
+	"regexp"
+	"strings"
+	"sync"
+
+	"golang.org/x/exp/maps"
+)
+
+type ProjectCollection struct {
+	Projects []*Project
+	mu       sync.Mutex
+}
+
+type Project struct {
+	Name     string
+	Findings []*Finding
+	Links    map[string]string
+	mu       sync.Mutex
+}
+
+func NewProject(name string) *Project {
+	return &Project{
+		Name:     name,
+		Findings: []*Finding{},
+		Links:    map[string]string{},
+	}
+}
+
+func NewProjectCollection() *ProjectCollection {
+	return &ProjectCollection{
+		Projects: []*Project{},
+	}
+}
+
+func normalizeProjectName(name string) string {
+	unacceptableChars := regexp.MustCompile(`[^\p{L}\p{N} \-\_]+`)
```

Comment:
[suggestion] can we add a comment explaining the intent of this regex in case there needs to be slight modifications in the future?

Reply:
Yup, we need to do better across the board. Regex patterns feel especially easy to break / misinterpret